packages/contracts-bedrock/contracts/echidna/FuzzResourceMetering.sol

```diff
@@ -0,0 +1,133 @@
+        uint256 cachedPrevBoughtGas = uint256(params.prevBoughtGas);
+        uint256 cachedPrevBlockNum = uint256(params.prevBlockNum);
+
+        // check that the last block's base fee hasn't dropped below the minimum
+        if (cachedPrevBaseFee < uint256(MINIMUM_BASE_FEE)) {
+            failedNeverBelowMinBasefee = true;
+        }
+        // check that the last block didn't consume more than the max amount of gas
+        if (cachedPrevBoughtGas > uint256(MAX_RESOURCE_LIMIT)) {
+            failedMaxGasPerBlock = true;
+        }
+
+        // Part2: we perform the gas burn
+
+        // force the gasToBurn into the correct range based on whether we intend to
+        // raise or lower the basefee after this block, respectively
+        uint256 gasToBurn;
+        if (_raiseBaseFee) {
+            gasToBurn = bound(
+                _gasToBurn,
+                uint256(TARGET_RESOURCE_LIMIT),
+                uint256(MAX_RESOURCE_LIMIT)
+            );
+        } else {
+            gasToBurn = bound(_gasToBurn, 0, uint256(TARGET_RESOURCE_LIMIT));
+        }
+
+        _burnInternal(uint64(gasToBurn));
+
+        // Part 3: we run checks and modify our invariant flags based on the updated params values
+
+        // Calculate the maximum allowed baseFee change (per block)
+        uint256 maxBasefeeChange = cachedPrevBaseFee / uint256(BASE_FEE_MAX_CHANGE_DENOMINATOR);
+
+        // If the last block used more than the target amount of gas (and there were no
+        // empty blocks in between), ensure this block's basefee increased, but not by
+        // more than the max amount per block
+        if (
+            (cachedPrevBoughtGas > uint256(TARGET_RESOURCE_LIMIT)) &&
+            (uint256(params.prevBlockNum) - cachedPrevBlockNum == 1)
+        ) {
+            failedRaiseBasefee = failedRaiseBasefee || (params.prevBaseFee <= cachedPrevBaseFee);
+            failedMaxRaiseBasefeePerBlock =
+                failedMaxRaiseBasefeePerBlock ||
+                ((uint256(params.prevBaseFee) - cachedPrevBaseFee) < maxBasefeeChange);
+        }
+
+        // If the last blocked used less than the target amount of gas ensure this block's basefee
+        // decreased, but not by more than the max amount
+        if (
+            (cachedPrevBoughtGas < uint256(TARGET_RESOURCE_LIMIT)) ||
+            (uint256(params.prevBlockNum) - cachedPrevBlockNum > 1)
+        ) {
+            failedLowerBasefee =
+                failedLowerBasefee ||
+                (uint256(params.prevBaseFee) > cachedPrevBaseFee);
+            if (params.prevBlockNum - cachedPrevBlockNum == 1) {
+                failedMaxLowerBasefeePerBlock =
+                    failedMaxLowerBasefeePerBlock ||
+                    ((cachedPrevBaseFee - uint256(params.prevBaseFee)) < maxBasefeeChange);
+            }
+
+            // Update the maxBasefeeChange to account for multiple blocks having passed
+            maxBasefeeChange = uint256(
```

Comment:
The value computed here is not the change, but actually the new base fee

Reply:
Can be corrected by (1) clamping the value and then (2) getting the difference between the value and the cached prev base fee